### ISSUE TITLE
wz-next="function", function is never executed

### ISSUE THREAD

**Issue by morissette** [mgonto/angular-wizard#148]
Description:
I am showing no execution of any function passed to wz-next, either in $scope.function or traditional function func() declarations. Both are simple functions that just log to the console yet nothing is logged. 

Angular: 1.4.6

Javascript, controller and module:
```javascript
(function() {
    var app = angular.module('mydhWizard', ['mgo-angular-wizard']);

    app.controller('WizardCtrl', function($scope, WizardHandler) {
        $scope.finished = function() {
            alert("Wizard finished :)");
        };

        $scope.logStep = function() {
            console.log("Step continued");
        };

        $scope.goBack = function() {
            WizardHandler.wizard().goTo(0);
        };

        $scope.setQuestionaire = function(num) {
            console.log('clicked');
        };
    });


})();
```

View and directive:
```html
<!-- load directive with wz-next function -->
            <wz-step title="Select Use Case">
                <h1 class="text-center">Select a Use Case</h1>
                <hr>
                <wizard-use-cases></wizard-use-cases>
            </wz-step>


<!-- the following is in the wizardUseCases directive template -->
<button type="button" class="btn btn-block btn-success" wz-next="setQuestionaire(1)">
  Get Started <i class="fa fa-chevron-right"></i>
</button>
```

**Comment by morissette** [comment]
Additionally it appears that wz-next cancels out ng-click methods

**Comment by morissette** [comment]
I don't like it but I went with the following solution while hoping for a more angularly way of solving my problem:

```javascript
    var app = angular.module('mydhWizard', ['mgo-angular-wizard']);

    app.controller('WizardCtrl', function($scope, WizardHandler) {
        $scope.finished = function() {
            alert("Wizard finished :)");
        };

        $scope.logStep = function() {
            console.log("Step continued");
        };

        $scope.goBack = function() {
            WizardHandler.wizard().goTo(0);
        };

        $scope.setQuestionaire = function(num) {
            console.log(num);
            WizardHandler.wizard().next();
        };
    });

    app.controller('WizardUseCaseCtrl', function($scope, $rootScope, WizardHandler) {
        var purposes = [
            "develop a new application in the cloud",
            "migrate an existing application to the cloud",
            "create a failover environement for my existing application",
            "create a cloud environment for development and testing",
            "create a cloud storage location for new or existing data",
            "archive data to long term or infrequently-accessed offsite storage",
            "create a disaster recovery environment",
        ];

        $scope.setQuestionaire = function(num, p) {
            $rootScope.wizard_cloud_purpose = "I want to " + purposes[p];
            $rootScope.wizard_questionaire = num;
            WizardHandler.wizard().next();
        };
    });
```

**Comment by jacobscarter** [comment]
So that is the way it's done, wzNext or any of the wz buttons do not take functions as arguments, they do not take arguments at all.  If you want custom action on a button hook it up to a ngClick (which it looks like you have) and finish it with the WizardHandler Service (as you have done).

**Comment by jacobscarter** [comment]
Also see #113

**Comment by jacobscarter** [comment]
@morissette I just pulled everything down and tested it out.  Using this syntax:

```html
<button wz-next="myFunction()">Click Me</button>
```

myFunction is called.  Also adding an ngClick to this button also calls the ngClick method, however it would be very bad practice to do this, especially if timing matters.  I would assume that your problem lies in the scope of your `<wizard-use-cases></wizard-use-cases>` directive but without seeing it I could say for sure.

Closing the issue, will re open if you have further questions
